### ISSUE TITLE
externalIPs Service Abstraction not preserving real client-ip even when using externalTrafficPolicy 

### ISSUE THREAD

**Issue by slecrenski** [kubernetes/kubernetes#66161]
**Is this a BUG REPORT or FEATURE REQUEST?**:  BUG REPORT

/kind bug
/sig network

**What happened**:

Background:

I am using Kubernetes externalIPs attribute in an nginx-ingress-controller service to bind to port 80/443 on each K8s Node.  This allows me to not really require a Load Balancer and use the Native features of Kubernetes.  Traffic can just ingress to any node and it is routed as necessary from that ingress point.  I have multiple ingress controllers running across the Cluster.

When looking at the ip address of incoming clients it appears to be either the pod ip, 127.0.0.1 or the host's default network eth0 ip.  When not using proxy protocol the ip is always 127.0.0.1.  When using proxy protocol the behavior is either the host's default network eth0 ip or the pod ip.

externalIPs and LoadBalancer/NodePort seem to be different concepts.  I am looking into using a NodePort instead of externalIPs but my environment firewalls are opened for 80/443 and NodePorts can only run on 30000+.  I am not sure how I could use the LoadBalancer concept.  I am creating an internal load balancer manually which routes traffic to each K8s node on 80/443.

As a related note the externalTrafficPolicy concept does not seem to apply to externalIPs.

**What you expected to happen**:

The client ip address should show as the real-ip of the user.

**How to reproduce it (as minimally and precisely as possible)**:

Setup Kubernetes Cluster
Deploy Canal aka Calico/Flannel with vxlan for networking
Deploy nginx-ingress-controller or not just use standard service with externalIPs set to the hostip of each Kubernetes Node.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.9.6
- Cloud provider or hardware configuration: N/A
- OS (e.g. from /etc/os-release): Centos 7
- Kernel (e.g. `uname -a`): Linux 3.10.0-693.21.1.el7.x86_64 #1 SMP Wed Mar 7 19:03:37 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
- Install tools: yum with Google Repos (Standard Install)
- Others:

**Comment by micw** [comment]
@slecrenski would you reopen this? it's still unresolved. thank you!

**Comment by slecrenski** [comment]
I ended up using NodePorts and deploying the nginx-ingress-controller/ingress-nginx as a daemonset.  The client-ip is preserved as long as the ingress-nginx is deployed across all of the nodes.  The load balancer just needs to get updated to point to the nodeport on all of the nodes.  Least that's how I did it.  Probably much more efficient ways to do it?  Probably wouldn't work with certain load balancer stickyness requirements but I would argue all services should always be deployed as stateless.
